address #8361 - introduce hook caller wrappers that enable backward compat (#8463)
pytest-dev/pytest: doc/en/deprecations.rst

:class:`PytestWarning` or subclasses, which can be filtered using :ref:`standard warning filters <warnings>`.

``py.path.local`` arguments for hooks replaced with ``pathlib.Path``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> In order to support the transition oto pathlib, the following hooks added additional arugments:

Reviewer: ```suggestion
In order to support the transition to :mod:`pathlib`, the following hooks now receive additional arguments:
```

> *  ``pytest_ignore_collect(fspath: pathlib.Path)``

Reviewer: ```suggestion
*  :func:`pytest_ignore_collect(fspath: pathlib.Path) <_pytest.hookspec.pytest_ignore_collect>`
```

> *  ``pytest_collect_file(fspath: pathlib.Path)``

Reviewer: *  :func:`pytest_collect_file(fspath: pathlib.Path) <_pytest.hookspec.pytest_collect_file>`

> *  ``pytest_pycollect_makemodule(fspath: pathlib.Path)``

Reviewer: ```suggestion
*  :func:`pytest_collect_file(fspath: pathlib.Path) <_pytest.hookspec.pytest_collect_file>`
```

> *  ``pytest_report_header(startpath: pathlib.Path)``

Reviewer: ```suggestion
*  :func:`pytest_collect_file(fspath: pathlib.Path) <_pytest.hookspec.pytest_collect_file>`
```

> *  ``pytest_report_collectionfinish(startpath: pathlib.Path)``

Reviewer: ```suggestion
*  :func:`pytest_collect_file(fspath: pathlib.Path) <_pytest.hookspec.pytest_collect_file>`
```

> The accompanying ``py.path.local`` based paths have been deprecated.

Reviewer: ```suggestion
The accompanying ``py.path.local`` based paths have been deprecated: plugins which manually invoke those hooks should only pass the new ``pathlib.Path`` arguments, and users should change their hook implementations to use the new ``pathlib.Path`` arguments.
```
